test_runner/src/main/kotlin/ftl/ios/xctest/common/ParseSwiftTests.kt

```diff
@@ -10,16 +14,24 @@ internal fun parseSwiftTests(binary: String): List<String> {
     // The OS limits the list of arguments to ARG_MAX. Setting the xargs limit avoids a fatal
     // 'argument too long' error. xargs will split the args and run the command for each chunk.
     // getconf ARG_MAX
-    val argMax = 262_144
+    // Windows has different limits
+    val argMax = if (isWindows) 251_36 else 262_144
 
-    val cmd = if (isMacOS) {
-        "nm -gU ${binary.quote()} | xargs -s $argMax xcrun swift-demangle"
-    } else {
-        "export LD_LIBRARY_PATH=~/.flank; export PATH=~/.flank:\$PATH; nm -gU ${binary.quote()} | xargs -s $argMax swift-demangle"
+    val cmd = when {
+        isMacOS -> "nm -gU ${binary.quote()} | xargs -s $argMax xcrun swift-demangle"
+        isLinux -> "export LD_LIBRARY_PATH=~/.flank; export PATH=~/.flank:\$PATH; nm -gU ${binary.quote()} | xargs -s $argMax swift-demangle"
+        isWindows -> "llvm-nm.exe --undefined-only --extern-only ${binary.quote().replace("\\", "/")} | xargs swift-demangle"
+        else -> throw RuntimeException("Unsupported OS for Integration Tests")
     }
 
+    val path = if (isWindows) {
+        listOf(Pair("Path", "$appDataDirectory\\.flank\\"), Pair("LD_LIBRARY_PATH", "$appDataDirectory\\.flank\\"))
+    } else emptyList()
+
     // https://github.com/linkedin/bluepill/blob/37e7efa42472222b81adaa0e88f2bd82aa289b44/Source/Shared/BPXCTestFile.m#L17-18
-    val demangledOutput = Bash.execute(cmd)
+    val shell = if (isWindows) ShellEnvironment.Cmd else ShellEnvironment.Default
+
+    val demangledOutput = Bash.execute(cmd, path, shell)
```

Comment:
would not be better to have executed function inside `ShellEnvironment` ?

Reply:
I think it is unnecessary as then we might as well create an entire new hierarchy. The sealed class is to help encapsulate the different shell environments and the Bash class simple executes what its been provided with. 
I understand that its the next logical step but if we go that far the entire class structure should be replaced entirely and reworked. :)
WDYT?

Reply:
we could leave it as it is, but for me 
`shell.execute(cmd, path)`
is much more readable than
`Bash.execute(cmd, path, shell)`

if you would like to leave it just resolve my comment

Reply:
@piotradamczyk5 I will create a ticket for refactoring this entire section. 
We need to look into how to handle commands and different shell environments.
https://github.com/Flank/flank/issues/1770